include/capo/pcm.hpp

```diff
@@ -35,4 +38,30 @@ struct PCM {
 	static Result<PCM> fromFile(std::string const& wavPath);
 	static Result<PCM> fromMemory(std::span<std::byte const> wavBytes);
 };
+
+class PCM::Streamer {
+  public:
+	Streamer();
+	Streamer(std::string path);
+	Streamer(PCM pcm);
+	Streamer(Streamer&&) noexcept;
+	Streamer& operator=(Streamer&&) noexcept;
+	~Streamer() noexcept;
+
+	Outcome open(std::string path);
```

Comment:
This is copy-and-move idiom. Since a copy is made internally (at least in the happy path), we let the compiler do that efficiently and then simply move it off the args list.